Gems/Atom/RPI/Code/Source/RPI.Public/MeshDrawPacket.cpp

```diff
@@ -422,6 +422,14 @@ namespace AZ
                         drawSrg->SetConstant(index, uvStreamTangentBitmask.GetFullTangentBitmask());
                     }
 
+                    // We need to call Reset for each ShaderItem, because not all shader have
+                    // the DrawSrg::m_modelLodMeshIndex constant defined in them.
+                    m_drawSrgModelLodMeshIndex.Reset();
+                    drawSrg->SetConstant(m_drawSrgModelLodMeshIndex, aznumeric_cast<uint32_t>(m_modelLodMeshIndex));
```

Comment:
Will this assert in debug builds if your shader does not have USE_DRAWSRG_MESHLOD_MESHINDEX enabled?

Reply:
No, it won't. When RHI::ShaderInputNameIndex is used with Srg::SetConstant it just returns false if it can't find the constant with given name.

Reply:
Btw, I just reverted usage of `RHI::ShaderInputNameIndex` for the shader constant `DrawSrg::m_modelLodMeshIndex` as it interferes with UnitTests:

```
Using RHI::ShaderInputNameIndex crashes the following
UnitTest MeshInstanceManagerTestFixture.AddInstance
because the NameDictionary singleton doesn't exist.

And even if I add NameDictionary::Create() in Setup()
and NameDictionary::Destroy() in Teardown() the
test fails due to a memory leak of 48 bytes.
```